staging/src/k8s.io/apiserver/pkg/server/options/authentication.go

```diff
@@ -195,9 +195,9 @@ type DelegatingAuthenticationOptions struct {
 	// before we fail the webhook call in order to limit the fan out that ensues when the system is degraded.
 	WebhookRetryBackoff *wait.Backoff
 
-	// ClientTimeout specifies a time limit for requests made by the authorization webhook client.
+	// TokenRequestTimeout specifies a time limit for requests made by the authorization webhook client.
 	// The default value is set to 10 seconds.
-	ClientTimeout time.Duration
+	TokenRequestTimeout time.Duration
```

Comment:
This reads as a potential breaking change to me. Are there any users/projects (both Kubernetes and external) that might use/set this value? Do we really have to change this in a patch release?

Reply:
> Are there any users/projects (both Kubernetes and external) that might use/set this value?

I'm not aware of any. The previous field was introduced in 1.20. I think most users rely on the default, which would mean they don't use this field directly.

Reply:
> This reads as a potential breaking change to me. Are there any users/projects (both Kubernetes and external) that might use/set this value? Do we really have to change this in a patch release?

Yes.  This bug actually increases kube-apiserver watch traffic for in-cluster authentication by 60x (10 seconds versus 10 minutes).  The fix here brings us back in line with 1.19 and reduces this traffic by 60x.

It is regrettable that we had the bug, but we need to fix it.

Reply:
@deads2k Thank you for the detailed explanation! I'm definitely okay with fixing this bug. I was referring to changing the field name because I'm not really sure should we rename a publicly exposed field in a patch release. That's why I asked are we aware of any users that might use/set `ClientTimeout` (e.g. other Kubernetes projects or folks building custom API servers).

To be sure that this change is okay, I already pinged other release managers to double-check how we handled this in the past. We'll get back to you as soon as possible.

Reply:
@xmudrii do you have any update on it?

Reply:
@p0lyn0mial Sorry for the delay! We've discussed this internally and we decided to move forward with those cherry-picks. I'll go ahead and approve them. Thank you for your patience!

Reply:
Thanks, Marko, could you please tag https://github.com/kubernetes/kubernetes/pull/101103 as well?

Reply:
@p0lyn0mial Already tagged #101103 as well. It's currently being retested, so it should be merged soon.